### ISSUE TITLE
Clarify Behavior of Negative TTL

### ISSUE THREAD

**Issue by shelley-jean-baker** [hazelcast/hazelcast#7729]
The documentation for map entry `ttl` [1,2] does not indicate the expected behavior if a negative value is provided:

> Puts an entry into this map with a given ttl (time to live) value. Entry will expire and get evicted after the ttl. If ttl is 0, then the entry lives forever.
> 
> ttl - maximum time for this entry to stay in the map 0 means infinite.

It is ambiguous as to whether this entry will immediately expire, live forever, or throw an exception. Some testing indicates that the entry will live forever (i.e. negative values seem to be functionally equivalent to zero).

```
public static void main(String[] args) throws Exception {
    HazelcastInstance instance = Hazelcast.newHazelcastInstance(new Config());
    IMap<String, Long> map = instance.getMap("map");
    set(map, "zero", 0);
    set(map, "negative", -10);
    set(map, "positive", 10);
    instance.shutdown();
}
private static void set(IMap<String, Long> map, String key, long ttl) {
    map.set(key, ttl, ttl, TimeUnit.SECONDS);
    EntryView<String, Long> view = map.getEntryView(key);
    System.out.println(String.format("%s=%s ttl=%d expiration=%s", 
            key, map.get(key), 
            view != null ? view.getTtl() : null,
            view != null ? new Date(view.getExpirationTime()) : null));
}
```

The javadoc should be updated to clarify this behavior.

[1] [`IMap.put(K, V, long, TimeUnit)`](http://docs.hazelcast.org/docs/3.6/javadoc/com/hazelcast/core/IMap.html#put%28K,%20V,%20long,%20java.util.concurrent.TimeUnit%29)
[2] [`IMap.set(K, V, long, TimeUnit)`](http://docs.hazelcast.org/docs/3.6/javadoc/com/hazelcast/core/IMap.html#set%28K,%20V,%20long,%20java.util.concurrent.TimeUnit%29)

**Comment by Serdaro** [comment]
@ahmetmircik, May I help on this one?

**Comment by Donnerbart** [comment]
The behavior seems to be a bit more complex, since there is also the option to configure a TTL in for the whole map (which has "live forever" as default). It should be like:
* positive TTL: is used, wins over the TTL from the `MapConfig`
* zero TTL: entry lives forever, wins over TTL from the `MapConfig`
* negative TTL: is ignored, TTL from the `MapConfig` is used

I created a PR with a test for these scenarios. I also created a PR to fix and cleanup the `IMap` JavaDoc.
